--- a/arch/arm/boot/dts/bcm47094-linksys-panamera.dts
+++ b/arch/arm/boot/dts/bcm47094-linksys-panamera.dts
@@ -266,3 +266,44 @@
 &usb3_phy {
 	status = "okay";
 };
+
+&nandcs {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot@0 {
+			label = "boot";
+			reg = <0x00000000 0x00080000>;
+			read-only;
+		};
+
+		nvram@80000 {
+			label = "nvram";
+			reg = <0x00080000 0x00100000>;
+		};
+
+		devinfo@180000 {
+			label = "devinfo";
+			reg = <0x00180000 0x00080000>;
+		};
+
+		firmware@200000 {
+			label = "firmware";
+			reg = <0x00200000 0x01d00000>;
+			compatible = "brcm,trx";
+		};
+
+		failsafe@1f00000 {
+			label = "failsafe";
+			reg = <0x01f00000 0x01d00000>;
+			compatible = "brcm,trx";
+		};
+
+		brcmnand@3c00000 {
+			label = "brcmnand";
+			reg = <0x03c00000 0x04400000>;
+		};
+	};
+};
